chap-rationale.tex

```diff
@@ -229,25 +226,23 @@ \section{Capability Register File}
   operating systems.
 \end{itemize}
 
-\noindent
-However, a design in
-which (more similar to the manner in which the 32-bit x86 ISA was extended to
-support 64-bit registers) the two files are \textit{merged} and able to hold
-both general-purpose integer registers and capability registers is
-also possible.
-This became a more appealing choice in the presence of 128-bit compressed
-capabilities, as register size doubled rather than quadrupled.
-
 Early in our design cycle, capability registers were able to hold only true
 capabilities (i.e., with tags); later, we weakened this requirement by adding
 an explicit tag bit to each register, in order to improve support for
 capability-oblivious code such as memory-copy routines able to copy data
 structures consisting of both capabilities and ordinary data.
-This shifted our approach somewhat more towards a merged approach; our view
-is that efficiency of implementation and compatibility (rather than
-maintaining a negligible
-effect on the software model) would be the primary reasons to select one
-approach or another for a particular starting-point ISA.
+
+With the separate register file on CHERI-MIPS, we also added
+instructions for copying non-capability data from a capability register
+into a general-purpose integer register. A use case for this was when a function was called
+with a parameter whose type is the union of a pointer and a non-pointer type,
+such as an int. This parameter had to be passed in a capability register, because
+the tag needed to be preserved when it held a capability. If the body of
+the function accessed the non-capability branch of the union, it needed to
+get the non-capability bits out of the capability register and into a general
+purpose register. This was originally done by spilling the capability register to the
+stack and then reading it back into a general-purpose integer register, but the register
+to register copy of \insnnoref{CGetAddr} proved faster.
```

Comment:
Do we want to mention the CSetAddr on the NULL register here as well?

Reply:
Perhaps.  This is definitely the place to note the use cases that motivated it.  I'm not sure I know that particular bit of history well enough to describe it.